asdf/tests/test_asdf.py

```diff
@@ -166,7 +166,6 @@ def test_asdf_file_version_requirement():
             af = AsdfFile(version="1.4.0", extensions=[extension_with_requirement])
 
 
-@pytest.mark.filterwarnings("ignore:unclosed file .*")
```

Comment:
Since this is fixing issue #1006 and it has a nice reproducer, I think it would also be a good idea to specifically create a regression test based on that reproducer, which documents that it is a regression test for fixing #1006.

Technically, this isn't necessary because some of the tests you are fixing the filters on do basically test for this, but those are side effects of the main purposes of those tests. Hence, we should specifically test for this behavior.